flink-java/src/main/java/org/apache/flink/api/java/RemoteEnvironment.java

```diff
@@ -133,26 +137,31 @@ public RemoteEnvironment(String host, int port, Configuration clientConfig,
 		this.port = port;
 		this.clientConfiguration = clientConfig == null ? new Configuration() : clientConfig;
 		if (jarFiles != null) {
-			this.jarFiles = new URL[jarFiles.length];
+			this.jarFiles = new ArrayList<URL>(jarFiles.length);
 			for (int i = 0; i < jarFiles.length; i++) {
 				try {
-					this.jarFiles[i] = new File(jarFiles[i]).getAbsoluteFile().toURI().toURL();
+					this.jarFiles.add(new File(jarFiles[i]).getAbsoluteFile().toURI().toURL());
 				} catch (MalformedURLException e) {
 					throw new IllegalArgumentException("JAR file path invalid", e);
 				}
 			}
 		}
 		else {
-			this.jarFiles = null;
+			this.jarFiles = Collections.emptyList();
+		}
+
+		if (globalClasspaths == null) {
+			this.globalClasspaths = Collections.emptyList();
+		} else {
+			this.globalClasspaths = Arrays.asList(globalClasspaths);
 		}
-		this.globalClasspaths = globalClasspaths;
 	}
 
 	// ------------------------------------------------------------------------
 
 	@Override
 	public JobExecutionResult execute(String jobName) throws Exception {
-		ensureExecutorCreated();
+		PlanExecutor executor = getExecutor();
```

Comment:
would it make sense to move the assignment to ```this.executor``` from ```getExecutor()``` to this line? as it stands the ScalaShellRemoteEnvironment executor is never closed.